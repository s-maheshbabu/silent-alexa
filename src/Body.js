import React, { Component } from "react";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import ChatWindow from "./ChatWindow";
import RightPanel from "./RightPanel";

export default class Body extends Component {
  render() {
    return [
      <MuiThemeProvider key="muiThemeProvider">
        <ChatWindow />
      </MuiThemeProvider>,

      <RightPanel key="rightPanel" />
    ];
  }
}
