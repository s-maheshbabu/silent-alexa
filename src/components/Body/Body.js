import React from "react";
import MuiThemeProvider from "material-ui/styles/MuiThemeProvider";
import ChatWindow from "ChatWindow/ChatWindow";
import RightPanel from "RightPanel/RightPanel";
import WelcomeScreen from "WelcomeScreen/WelcomeScreen";
import { useCookies } from "react-cookie";

import { AMAZON_LOGIN_COOKIE } from "Constants";

export default function Body() {
  const [cookies] = useCookies([AMAZON_LOGIN_COOKIE]);

  if (cookies[AMAZON_LOGIN_COOKIE] !== undefined) {
    return [
      <MuiThemeProvider key="muiThemeProvider">
        <ChatWindow />
      </MuiThemeProvider>,

      <RightPanel key="rightPanel" />
    ];
  }

  return <WelcomeScreen />;
}
