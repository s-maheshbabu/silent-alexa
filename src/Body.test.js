import React from "react";
import { shallow, mount } from "enzyme";
import Body from "./Body";

const mockIsAuthenticationInfoValid = jest.fn();

beforeEach(() => {
  jest.resetAllMocks();
});

it("renders Body when isAuthenticationInfoValid returns true", () => {
  mockIsAuthenticationInfoValid.mockReturnValueOnce(true);
  const wrapper = shallow(
    <Body isAuthenticationInfoValid={mockIsAuthenticationInfoValid} />
  );
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});

it("renders Body when isAuthenticationInfoValid returns false", () => {
  mockIsAuthenticationInfoValid.mockReturnValueOnce(false);
  const wrapper = shallow(
    <Body isAuthenticationInfoValid={mockIsAuthenticationInfoValid} />
  );
  expect(wrapper).toMatchSnapshot();

  wrapper.unmount();
});

// TODO: We should use shallow or render here instead of mount. However, because
// Body is a component that returns an array of elements, enzyme doesn't
// yet support this feature.

// They fixed it here https://github.com/airbnb/enzyme/commit/6f7f4fbae19050a912c70e6fbfe42c47b0851fe3

// It is not yet released on npm though. Once enzyme version > 3.3.0 is
// released, we need to fix this test.
it("verifies that authenticationInfo is passed to ChatWindow component", () => {
  mockIsAuthenticationInfoValid.mockReturnValueOnce(true);
  const authenticationInfo = {};
  const body = mount(
    <Body
      isAuthenticationInfoValid={mockIsAuthenticationInfoValid}
      authenticationInfo={authenticationInfo}
    />
  );

  // Verify that ChatWindow recieves authenticationInfo prop
  const authenticationInfoProp = body
    .find("ChatWindow")
    .prop("authenticationInfo");
  expect(authenticationInfoProp).toBe(authenticationInfo);
});
