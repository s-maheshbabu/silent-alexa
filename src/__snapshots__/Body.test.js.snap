// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without crashing 1`] = `
Array [
  <MuiThemeProvider
    key="muiThemeProvider"
/>,
  <RightPanel
    key="rightPanel"
/>,
]
`;
